resources/channel.sub.js

```diff
@@ -171,19 +192,17 @@
             if (typeof fn !== "function") {
                 throw new TypeError(`Expected function, got ${typeof fn}`);
             }
-            if (!this.eventListeners.has(type)) {
-                this.eventListeners.set(type, new Set());
+            if (!this.eventListeners.hasOwnProperty(type)) {
+                throw new Error(`Unrecognised event type ${type}`);
```

Comment:
Alright, this settles the questions I had in the previous review about unnecessary (now necessary) code to add empty event handler maps. Thanks!